scripts/tasks/ts.ts

```diff
@@ -1,48 +1,89 @@
+import fs from 'fs';
 import * as path from 'path';
-import { tscTask, argv } from 'just-scripts';
+import { tscTask, argv, TscTaskOptions, resolveCwd, logger } from 'just-scripts';
+import { Arguments } from 'yargs';
+
+interface JustArgs extends Arguments {
+  production?: boolean;
+}
 
 const libPath = path.resolve(process.cwd(), 'lib');
 const srcPath = path.resolve(process.cwd(), 'src');
 // Temporary hack: only use tsbuildinfo file for things under packages/fluentui
 const useTsBuildInfo =
   /[\\/]packages[\\/]fluentui[\\/]/.test(process.cwd()) && path.basename(process.cwd()) !== 'perf-test';
 
-function getExtraTscParams(args) {
+/**
+ *
+ * Explicitly set `baseUrl` to current package root for packages (converged packages) that use TS path aliases.
+ * > - This is a temporary workaround for current way of building packages.
+ * > - Without setting baseUrl we would get all aliased packages build within outDir
+ */
+function backportTsAliasedPackages(options: TscTaskOptions) {
+  const tsConfigFilePath = resolveCwd('./tsconfig.json');
+  const tsConfig = JSON.parse(fs.readFileSync(tsConfigFilePath, 'utf-8'));
```

Comment:
JSON.parse won't work if the tsconfig has comments. There's a helper in `../read-config` which handles comments.

Reply:
I know. comments are not allowed in json unless we use json5 everywhere and all 3rd party tools as well. I haven't seen need to provide comments within tsconfigs in particular in my whole carer (although I need to point out that `tsc --init` creates this huge "documented" tsconfig.json for "learning purposes" - that's not our case.

Also using jju to slow down build pipeline ,just to be able to have comments in self explanatory configs, doesn't feel like good solution to me.

Reply:
Up to this point we've supported comments in tsconfig in this repo, and it's a common enough convention that all (or almost all) tools I've encountered which interact with tsconfigs support it. In the handful of cases where it's not self-explanatory why a certain setting is needed, the ability to provide comments is extremely helpful for maintainability. In any case I can think of involving tsconfig parsing, the slightly increased runtime cost of a parser that supports comments is not going to be the bottleneck.

Reply:
> slightly increased runtime cost of a parser that supports comments is not going to be the bottleneck.

we don't have any data unfortunately. We should measure first.

Reply:
> > slightly increased runtime cost of a parser that supports comments is not going to be the bottleneck.
> 
> we don't have any data unfortunately. We should measure first.

Thanks for adding the comment support. Agreed in general that measurement is good, but here, the thing that would need justification by measurement (unless we had very strong reason to suspect it being a bottleneck) is the argument that we should *not* support comments out of perf concerns, given the negative impact on maintainability and deviation from community norms.